File: Ruby/BubbleSort.rb
#Bubble Sort is the simplest sorting algorithm. It works by repeatedly swapping the adjacent elements if they are in wrong order.
#The time Complexity of Bubble Sort algorithm is O(N^2) where N is the number of elements in the array.

def bubbleSort (array, size)
  array.each do
    for j in 0 .. (size-2)
      if array[j] > array[j+1]
        temp = array[j]
        array[j] = array[j+1]
        array[j+1] = temp
      end
    end
  end
  array
end


array = [0,9,1,4,2]

print(array)

bubbleSort(array, array.length)

puts("\nAfter sorting: ")

print(array)

#Output
#    [0, 9, 1, 4, 2]
#    After Sorting:
#    [0, 1, 2, 4, 9]

File: Java/BubbleSort.java
public class Bubble {

    public static void main(String[] args) {
        Bubble ob1 = new Bubble();
        int arr[] = {64, 34, 25, 12, 22, 11, 90};
        ob1.bubbleSort(arr);
        System.out.println("This is the Sorted array");
        ob1.displayArray(arr);
    }

    void bubbleSort(int arr[]) {
        int a = arr.length;
        for (int i = 0; i < a - 1; i++) {
            for (int j = 0; j < a - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    void displayArray(int arr[]) {
        int a = arr.length;
        for (int i = 0; i < a; ++i) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
}